### PR TITLE
perf(web): ~400x faster http header trimming

### PR DESCRIPTION
**Pull request by AaronO**
Use a regex substring match with a first/last char fastpath instead of 2 regex replaces. Roughly ~400x faster (423ms vs 0.7ms in profiled runs)

### CODE REVIEW

**Comment by aapoalas** [comment]
LGTM. Only possible improvement I could see is to check do the substring manually as well. Something like this:

```javascript
const needsTrimStart = isHttpWhitespace(str[0]);
const needsTrimEnd = isHttpWhitespace(str[str.length - 1]);
if (!needsTrimStart && !needsTrimEnd) return str;

let startIndex = Number(needsTrimStart);
if (needsTrimStart) {
  startIndex = getFirstNonHttpWhitespaceIndex(startIndex);
}
let endIndex = needsTrimEnd ? getLastNonHttpWhitespaceIndex(str.length - 2) : undefined;
return str.substring(startIndex, endIndex);
```

**Comment by AaronO** [comment]
> LGTM. Only possible improvement I could see is to check do the substring manually as well. Something like this:
> 
> ```js
> const needsTrimStart = isHttpWhitespace(str[0]);
> const needsTrimEnd = isHttpWhitespace(str[str.length - 1]);
> if (!needsTrimStart && !needsTrimEnd) return str;
> 
> let startIndex = Number(needsTrimStart);
> if (needsTrimStart) {
>   startIndex = getFirstNonHttpWhitespaceIndex(startIndex);
> }
> let endIndex = needsTrimEnd ? getLastNonHttpWhitespaceIndex(str.length - 2) : undefined;
> return str.substring(startIndex, endIndex);
> ```

Had considered that too, but I'm happy mainly optimizing for the fast-path of no trimming right now since that should be the the dominant path too

**Comment by aapoalas** [comment]
> Had considered that too, but I'm happy mainly optimizing for the fast-path of no trimming right now since that should be the the dominant path too

Makes sense.

**Review by LePichu — APPROVED** [review]

**Review by ry — APPROVED** [review]
LGTM

File: ext/fetch/20_headers.js [hunk 1]
```diff
@@ -59,17 +58,7 @@
...
-      HTTP_WHITESPACE_PREFIX_RE,
-      "",
-    );
-    potentialValue = StringPrototypeReplaceAll(
-      potentialValue,
-      HTTP_WHITESPACE_SUFFIX_RE,
-      "",
-    );
-    return potentialValue;
+    return httpTrim(potentialValue);
```
Comment (Genarito) [inline ext/fetch/20_headers.js:61]: Hi! Sorry about my ignorance, but why not just move the entire content of `httpTrim` to `normalizeHeaderValue` function? That'd remove `httpTrim` definition everywhere which is a lot of lines less with the same result

**Pull request merged**